tools/src/launcher/Main.hs

```diff
@@ -338,13 +346,31 @@ main =
                     loNodeLogConfig
                     (NodeData loNodePath realNodeArgs loNodeLogPath)
                     (NodeData wpath loWalletArgs loWalletLogPath)
-                    (UpdaterData
-                        loUpdaterPath loUpdaterArgs loUpdateWindowsRunner loUpdateArchive)
+                    (setUpdaterData lo)
                     loNodeTimeoutSec
                     loReportServer
                     loWalletLogging
                 logNotice "Finished clientScenario"
   where
+    setUpdaterData :: LauncherOptions -> UpdaterData
+    setUpdaterData LO {..} = case buildOS of 
+        OSX -> OSXUpdaterData OSXUD
+            { oudOpenPath = loUpdaterPath
+            , oudArgs = loUpdaterArgs
+            -- It would be better to use `bug` function from universum-1.0.1 but you want 0.9.0
+            -- , oudPkgPath = fromMaybe (bug Not_set_OSX_loUpdateArchive) loUpdateArchive
+            , oudPkgPath = fromJust loUpdateArchive
```

Comment:
See [exception handling guidelines](https://github.com/input-output-hk/cardano-sl/blob/37ab33599470b30c608528d5a844a3a59ba9843f/docs/exceptions.md). Functions like `fromJust` and `bug` should be used only if their failure implies a programmer mistake. But here programmer can't guarantee that launcher config will have `loUpdateArchive` when launcher is launched on OSX. The config is passed as a command line option, so anything can be passed there.
Consider changing `setUpdaterData` type to `LauncherOptions -> IO UpdaterData` (and use `throwM` if something goes wrong).